fairscale/nn/data_parallel/fully_sharded_data_parallel.py

```diff
@@ -1001,6 +1001,10 @@ def _set_is_root(self) -> None:
             return
         # No FullyShardedDataParallel instance wraps this, else _is_root would be set to False.
         self._is_root = True
+        # If final backward callback is never been queued, state should be IDLE.
+        # If final backward callback is queued, the callback should be finished
+        # and the state was reset to be IDLE.
```

Comment:
Can you add a comment on why asserting on root only? It is good to assert on root only, but good to know why it is the case.